spec/lib/training_route_manager_spec.rb

```diff
@@ -3,103 +3,93 @@
 
           it "returns true" do
-            expect(subject.requires_schools?).to be true
+            expect(subject).to be true
           end
         end
 
-        context "with the :routes_#{route} feature flag disabled", "feature_routes.#{route}": false do
-          context "with a non school direct trainee" do
-            let(:trainee) { build(:trainee) }
+        context "trainee on a non #{training_route_enums_key} training route " do
+          let(:trainee) { build(:trainee, other_training_route || :assessment_only) }
 
-            it "returns false" do
-              expect(subject.requires_schools?).to be false
-            end
+          it "returns false" do
+            expect(subject).to be false
           end
         end
       end
-    end
-  end
 
-  describe "#requires_employing_school?" do
-    context "with the :routes_school_direct_salaried feature flag enabled", "feature_routes.school_direct_salaried": true do
-      context "with a school direct salaried trainee" do
-        let(:trainee) { build(:trainee, :school_direct_salaried) }
+      context "with the :routes_#{training_route_enums_key} feature flag disabled", "feature_routes.#{training_route_enums_key}": false do
+        context "trainee on a #{training_route_enums_key} training route" do
+          let(:trainee) { build(:trainee, training_route: training_route_enums_key) }
 
-        it "returns true" do
-          expect(subject.requires_employing_school?).to be true
+          it "returns false" do
+            expect(subject).to be false
+          end
         end
-      end
 
-      context "with a non school direct trainee" do
-        let(:trainee) { build(:trainee) }
+        context "trainee on a non #{training_route_enums_key} training route " do
+          let(:trainee) { build(:trainee, other_training_route || :assessment_only) }
 
-        it "returns false" do
-          expect(subject.requires_employing_school?).to be false
+          it "returns false" do
+            expect(subject).to be false
+          end
         end
       end
     end
-
-    context "with the :routes_school_direct_salaried feature flag disabled", "feature_routes.school_direct_salaried": false do
-      let(:trainee) { build(:trainee) }
-
-      it "returns false" do
-        expect(subject.requires_employing_school?).to be false
-      end
-    end
   end
 
-  describe "#early_years_route?" do
-    context "non early years route" do
-      let(:trainee) { build(:trainee, :school_direct_salaried) }
-
-      it "returns false" do
-        expect(subject.early_years_route?).to be false
-      end
-    end
-
-    context "early years route" do
-      let(:trainee) { build(:trainee, :early_years_undergrad) }
-
-      it "returns true" do
-        expect(subject.early_years_route?).to be true
-      end
-    end
-  end
+  include_examples "enabled? checks", "requires_placement_details?", "provider_led_postgrad"
```

Comment:
I find these really hard to follow. Might just be me obvs but I prefer the duplication in specs as it makes them easier to read. Doesn't need to block this PR but we should see what the rest of the team think and add whatever we decide to the testing guidelines doc we have.

Reply:
We discussed this offline and the team decided that in general we should avoid shared examples and test explicitly. Not an absolute rule if shared examples make something easier to understand though.

Reply:
redone as https://github.com/DFE-Digital/register-trainee-teachers/pull/1195/commits/0f477c0f51d992a544ca52497bdb9b6ba75b6cf4